src/textual/renderables/text_opacity.py

```diff
@@ -63,6 +63,7 @@ def process_segments(
         _from_color = Style.from_color
         if opacity == 0:
             for text, style, control in segments:
+                assert style is not None
```

Comment:
Technically correct, but slow. There may be a lot of segments. Suggest ignoring or casting segments to say that style will not be None.

Reply:
I'll do that latter.

Reply:
You can remove the assert now.

Reply:
Oops :) Removed.